### PR TITLE
Implement the INVERT_ZERO_DC flag in LaplaceCyclic

### PR DESCRIPTION
**Pull request by johnomotani**
I've labelled as a bugfix because previously the flag was ignored silently. The `INVERT_ZERO_DC` flag is useful when combining a `Laplacian` solver with a `LaplaceXY` solver which calculates the DC part of the potential.

It would be nice to check that the flags passed by the user have been implemented, but I don't have time to check all the flags now. See #1785.

### CODE REVIEW

**Comment by bendudson** [comment]
Thanks @johnomotani, just wondering why `startz` is an `int` rather than a `bool`.

**Comment by johnomotani** [comment]
I'd used an `int` because when I was writing the first place I put it was `int kz = startz`, but `bool startz` would work. Would you prefer a `bool`?

File: src/invert/laplace/impls/cyclic/cyclic_laplace.cxx [hunk 1]
```diff
@@ -232,9 +232,19 @@ FieldPerp LaplaceCyclic::solve(const FieldPerp& rhs, const FieldPerp& x0) {
       auto k1d = Array<dcomplex>((localmesh->LocalNz) / 2 +
                                  1); // ZFFT routine expects input of this length
 
+      bool zero_DC = false;
+      if(global_flags & INVERT_ZERO_DC) {
+        // No DC component
+        zero_DC = true;
```
Comment (ZedThree) [inline src/invert/laplace/impls/cyclic/cyclic_laplace.cxx]: Could do `const bool zero_DC = global_flags & INVERT_ZERO_DC;`

File: src/invert/laplace/impls/cyclic/cyclic_laplace.cxx [hunk 2]
```diff
@@ -440,13 +446,23 @@ Field3D LaplaceCyclic::solve(const Field3D& rhs, const Field3D& x0) {
       auto k1d = Array<dcomplex>((localmesh->LocalNz) / 2 +
                                  1); // ZFFT routine expects input of this length
 
+      bool zero_DC = false;
```
Comment (ZedThree) [inline src/invert/laplace/impls/cyclic/cyclic_laplace.cxx]: Sorry, I meant to flag this one too!

**Pull request merged**